daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerMode.scala

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.script
+
+import java.io.File
+
+sealed abstract class LedgerMode extends Product with Serializable
+object LedgerMode {
+  final case class LedgerAddress(host: String, port: Int) extends LedgerMode
+  final case class ParticipantConfig(file: File) extends LedgerMode
+  final case class IdeLedger() extends LedgerMode
+}
```

Comment:
Nice.

Reply:
Thoughts on the names btw? I was a little unsure, as I'm not super familiar with what a participant config actually is. Especially given the first LedgerAddress case just builds a Participant of the same type as ParticipantConfig

Reply:
Not fantastic: naming is hard. 
I think it is ok if you always use them with prefix (`LedgerMode.XXX`). 
maybe LedgerConfig is better ?

anyway it is better now than before.

Reply:
I'll give them another think, to be sure
Maybe `RemoteParticipantHost`, `RemoteParticipantConfig` and `IdeLedgerParticipant`